### PR TITLE
prov/efa: doxygen doc setup and examples

### PR DESCRIPTION
**Pull request by wzamazon**
setup doxygen config file, and document `struct rxr_pkt_entry` as an example

### CODE REVIEW

File: prov/efa/src/rxr/rxr_pkt_entry.h [hunk 1]
```diff
@@ -65,24 +70,83 @@ struct rxr_pkt_sendv {
...
+ * For each request EFA provider submits to (EFA device or SHM), it will allocate a packet entry.
+ * 
+ * When the request was submitted, the pointer of the packet entry will be used as context.
+ * For EFA device, context is work request ID (`wr_id`), For SHM, context is the `op_context`
+ * in a `fi_msg`.
+ * 
+ * When the request was completed, EFA device or SHM will return a completion entry, with the
+ * the pointer to the rxr_pkt_entry in it.
+ * 
+ * In a specail occasion, he completion can be a Receiver Not Ready (RNR) error completion.
```
Comment (tschuett) [inline prov/efa/src/rxr/rxr_pkt_entry.h:89]: special occasion, the completion

File: prov/efa/src/rxr/rxr_pkt_entry.h [hunk 2]
```diff
@@ -65,24 +70,83 @@ struct rxr_pkt_sendv {
...
+ * 
+ * For EFA device, a TX operation can be send/read.
+ * 
+ * For SHM, a TX operation can be send/read/write/atomic.
+ * 
+ * When used as context of request, packet was allocated from endpoint's shm/efa_tx/rx_pool.
+ * When the request is to EFA device, the packet's memory must be registered to EFA device,
+ * and the memory registration must be stored as the `mr` field of packet entry.
+ * 
+ * Second, packet entries can be used o store received packet entries that is
```
Comment (tschuett) [inline prov/efa/src/rxr/rxr_pkt_entry.h:103]: o store?

File: prov/efa/src/rxr/rxr_pkt_entry.h [hunk 3]
```diff
@@ -65,24 +70,83 @@ struct rxr_pkt_sendv {
...
+ * unexpected or out-of-order. This is because the efa/shm_rx_pkt_pool's size is fixed,
+ * therefore cannot be used to hold unexpected/out-of-order packets. When an unexpected/out-of-order
```
Comment (tschuett) [inline prov/efa/src/rxr/rxr_pkt_entry.h:105]: it cannot

File: prov/efa/src/rxr/rxr_pkt_entry.h (hunk 1 above)
Comment (wzamazon) [reply prov/efa/src/rxr/rxr_pkt_entry.h]: fixed

File: prov/efa/src/rxr/rxr_pkt_entry.h (hunk 2 above)
Comment (wzamazon) [reply prov/efa/src/rxr/rxr_pkt_entry.h]: fixed

File: prov/efa/src/rxr/rxr_pkt_entry.h (hunk 3 above)
Comment (wzamazon) [reply prov/efa/src/rxr/rxr_pkt_entry.h]: fixed

**Review by wckzhang — APPROVED** [review]

**Pull request merged**